haystack/preview/components/rankers/similarity.py

```diff
@@ -0,0 +1,110 @@
+import logging
+from pathlib import Path
+from typing import List, Optional, Union, Dict, Any
+
+from haystack.preview import ComponentError
+
+from haystack.lazy_imports import LazyImport
+from haystack.preview import Document, component, default_from_dict, default_to_dict
+
+
+logger = logging.getLogger(__name__)
+
+
+with LazyImport(message="Run 'pip install transformers[torch,sentencepiece]==4.32.1'") as torch_and_transformers_import:
+    import torch
+    from transformers import AutoModelForSequenceClassification, AutoTokenizer
+
+
+@component
+class SimilarityRanker:
+    """
+    Ranks documents based on query similarity.
+
+    Usage example:
+    ```
+    from haystack.preview import Document
+    from haystack.preview.components.rankers import SimilarityRanker
+
+    sampler = SimilarityRanker()
+    docs = [Document(text="Paris"), Document(text="Berlin")]
+    query = "City in Germany"
+    output = sampler.run(query=query, documents=docs)
+    docs = output["documents"]
+    assert len(docs) == 2
+    assert docs[0].text == "Berlin"
+    ```
+    """
+
+    def __init__(
+        self,
+        model_name_or_path: Union[str, Path] = "cross-encoder/ms-marco-MiniLM-L-6-v2",
```

Comment:
I would just go with `model` to be fair, type hints and documentation can do the rest.